subscription-service/src/main/kotlin/com/egm/stellio/subscription/job/TimeIntervalNotificationJob.kt

```diff
@@ -0,0 +1,80 @@
+package com.egm.stellio.subscription.job
+
+import com.egm.stellio.shared.model.JsonLdEntity
+import com.egm.stellio.shared.model.Notification
+import com.egm.stellio.shared.util.JsonLdUtils
+import com.egm.stellio.shared.util.JsonUtils
+import com.egm.stellio.shared.util.toUri
+import com.egm.stellio.subscription.model.EntityInfo
+import com.egm.stellio.subscription.model.Subscription
+import com.egm.stellio.subscription.service.NotificationService
+import com.egm.stellio.subscription.service.SubscriptionService
+import kotlinx.coroutines.reactive.awaitFirst
+import kotlinx.coroutines.runBlocking
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.scheduling.annotation.EnableScheduling
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Component
+import org.springframework.web.reactive.function.client.WebClient
+
+@EnableScheduling
+@Component
+class TimeIntervalNotificationJob(
+    private val subscriptionService: SubscriptionService,
+    private val notificationService: NotificationService,
+    @Autowired
+    private val webClient: WebClient
+) {
+
+    @Scheduled(fixedRate = 1000)
+    fun sendTimeIntervalNotification() {
+        runBlocking {
+            subscriptionService.getRecurringSubscriptionsToNotify().forEach {
+                sendNotification(getEntitiesToNotify(it.entities, it.q), it)
+            }
+        }
+    }
+
+    fun extractParam(entityInfo: EntityInfo, q: String?): String {
+        val param = java.lang.StringBuilder()
+        param.append("?type=${entityInfo.type}")
+        if (entityInfo.id != null) param.append("&id=${entityInfo.id}")
+        if (entityInfo.idPattern != null) param.append("&idPattern=${entityInfo.idPattern}")
+        if (q != null) param.append("&q=$q")
+        return param.toString()
+    }
+
+    suspend fun sendNotification(listEntity: List<JsonLdEntity>, subscription: Subscription):
```

Comment:
add a TODO here because you should actually send only one notification per subscription